csharp/src/Ice/UriParser.cs

```diff
@@ -23,95 +31,50 @@ internal static class UriParser
             GenericUriParserOptions.IriParsing |
             GenericUriParserOptions.NoUserInfo;
 
-        /// <summary>Checks if a string is an ice or ice+transport URI and not a stringified proxy using the old format.
-        /// </summary>
+        /// <summary>Checks if a string is an ice or ice+transport URI, and not a proxy or endpoint string using the
+        /// ice1 string format.</summary>
         /// <param name="s">The string to check.</param>
         /// <returns>True when the string is most likely an ice or ice+transport URI; otherwise, false.</returns>
         internal static bool IsUri(string s) =>
             (s.StartsWith("ice:") || (s.StartsWith("ice+") && s.Contains(':'))) && !s.Contains(' ');
 
         /// <summary>Parses an ice or ice+transport URI string.</summary>
         /// <param name="uriString">The URI string to parse.</param>
-        /// <param name="oaEndpoints">When true, the URI is used to configure an object adapter.</param>
         /// <param name="communicator">The communicator.</param>
-        /// <returns>The parsed components of the ice or ice+transport URI.</returns>
-        internal static (Encoding Encoding,
-                         IReadOnlyList<Endpoint> Endpoints,
-                         string Facet,
-                         InvocationMode InvocationMode,
-                         List<string> Path,
-                         Protocol Protocol) Parse(
+        /// <param name="proxyOptions">When not null, the uriString represents a proxy and this method parses proxy
+        /// options into this parameter.</param>
+        /// <returns>The Uri and endpoints of the ice or ice+transport URI.</returns>
+        internal static (Uri Uri, IReadOnlyList<Endpoint> Endpoints) Parse(
             string uriString,
-            bool oaEndpoints,
-            Communicator communicator)
+            Communicator communicator,
+            ProxyOptions? proxyOptions = null)
```

Comment:
I find this proxyOptions a bit confusing... it looks like it's both an input (is OA URI or Proxy URI) and output parameter (if ProxyURI, it returns the protocol/encoding). Providing 2 separate methods for proxy URI/OA URI parsing with return different results might be less confusing?

Reply:
I think it's cleaner to have a single `Parse` method on `UriParser`, and use this nullable proxyOptions to select between proxy parsing and endpoint parsing.
There is no good name for multiple Parse "overloads": ParseForReferenceCreation? ParseIntoEndpoints?

Reply:
I changed my mind, and I will fix this in my next PR.